packages/typescript-estree/src/node-utils.ts

```diff
@@ -663,3 +663,27 @@ export function firstDefined<T, U>(
   }
   return undefined;
 }
+
+export function identifierIsThisKeyword(id: ts.Identifier): boolean {
+  return id.originalKeywordKind === SyntaxKind.ThisKeyword;
+}
+
+export function isThisIdentifier(node: ts.Node | undefined): boolean {
```

Comment:
you should use TsNode type and remove casts

Reply:
you should use a type guard so that TS can understand the refinement you're making within the  function

```suggestion
export function isThisIdentifier(node: ts.Node | undefined): node is ts.ThisKeyword {
```

Reply:
`ts` does not have nodes named `ThisKeyword`. The most similar thing is `ThisExpression`. Probably not what we want.
```ts
export interface ThisExpression extends PrimaryExpression {
    readonly kind: SyntaxKind.ThisKeyword;
}
```